src/components/split-button/split-button.pw.tsx

```diff
@@ -0,0 +1,874 @@
+    await mount(
+      <SplitButtonList data-element="split-button-cypress-element" />
+    );
+
+    await expect(splitMainButton(page)).toHaveAttribute(
+      "data-element",
+      "split-button-cypress-element"
+    );
+  });
+
+  test(`should render with data-role prop`, async ({ mount, page }) => {
+    await mount(<SplitButtonList data-role="split-button-cypress-role" />);
+
+    await expect(splitMainButton(page)).toHaveAttribute(
+      "data-role",
+      "split-button-cypress-role"
+    );
+  });
+
+  (["left", "right"] as SplitButtonProps["align"][]).forEach((alignment) => {
+    test(`should render with alignment to the ${alignment}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<SplitButtonList align={alignment} />);
+
+      await expect(splitMainButton(page)).toHaveCSS(
+        `margin-${alignment}`,
+        "0px"
+      );
+    });
+  });
+
+  test(`should render with component disabled`, async ({ mount, page }) => {
+    await mount(<SplitButtonList disabled />);
+
+    await mainButton(page).hover();
+    await expect(
+      splitMainButton(page).locator("button").first()
+    ).toBeDisabled();
+  });
+
+  ([
+    ["after", "left"],
+    ["before", "right"],
+  ] as [SplitButtonProps["iconPosition"], string][]).forEach(
+    ([iconPosition, margin]) => {
+      test(`should render with icon position ${iconPosition}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(
+          <SplitButtonList iconType="add" iconPosition={iconPosition}>
+            IconPosition
+          </SplitButtonList>
+        );
+
+        await expect(icon(page).first()).toHaveCSS(`margin-${margin}`, "8px");
+      });
+    }
+  );
+});
+
+test.describe("Events tests", () => {
+  test(`should click a main element of SplitButton component`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <SplitButtonList
+        onClick={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await splitMainButtonDataComponent(page, 0).click();
+    expect(callbackCount).toBe(1);
+  });
+});
+
+test.describe("Functional tests", () => {
+  test(`should render component and expand by hovering`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SplitButtonList />);
+
+    await getDataElementByValue(page, "dropdown").hover();
+    await expect(additionalButton(page, 0)).toBeVisible();
+    await expect(additionalButton(page, 1)).toBeVisible();
+    await expect(additionalButton(page, 2)).toBeVisible();
+    await expect(splitToggleButton(page)).toHaveAttribute(
+      "aria-expanded",
+      "true"
+    );
+  });
+
+  test(`should render component in a hidden container and expand by hovering`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <Accordion title="Heading">
+        <SplitButtonList />
+      </Accordion>
+    );
+
+    await accordionDefaultTitle(page).click();
```

Comment:
**suggestion**: in the original test this seemed to be checking enter key behaviour, not clicking. I know they're very similar in terms of result but maybe it should be worth checking both here?